src/ocl_utils_mod.f90

```diff
@@ -280,19 +307,46 @@ end subroutine release_kernel
 
   !> Set-up the specified number of OpenCL command queues for the specified
   !! context and device.
-  subroutine init_cmd_queues(nqueues, queues, context, device)
+  subroutine init_cmd_queues(nqueues, queues, context, device, &
+                             enable_profiling, out_of_order)
     !> The number of command queues to create
     integer, intent(in) :: nqueues
     integer(c_intptr_t), target, intent(inout) :: queues(nqueues)
     integer(c_intptr_t), intent(in) :: device
     integer(c_intptr_t), intent(in) :: context
+    logical, intent(in), optional :: enable_profiling, out_of_order
     ! Locals
     integer :: i
+    logical :: profiling_property, out_of_order_property
     integer(c_int32_t) :: ierr
+    integer(c_int64_t), parameter :: no_properties = b'00'
```

Comment:
Can these bit masks be constructed from the OpenCL parameters themselves (CL_QUEUE_PROFILING_ENABLE, etc.) rather than hardwiring their values?

Reply:
I didn't know about Fortran bitwise operators, it looks much nicer now :)